app/models/work_package/pdf_export/export/markdown.rb

```diff
@@ -94,22 +94,27 @@ def handle_wp_mention_html_tag(tag, node, opts)
         # clear the text content, so it does not get rendered
         next_node.string_content = ""
       end
-      wp_mention_macro(tag.attr("data-text") || "", id, opts)
+      wp_mention_macro(tag.attr("data-text") || "", id.delete("#"), opts)
```

Comment:
Maybe better to follow how it is done in `OpenProject::TextFormatting::Filters::MentionFilter` and use something like `id[/\d+/]`

Reply:
Agreed, it is more robust. 

fixed in https://github.com/opf/openproject/pull/19490/commits/61944a5409b58034c7ac0baec8fcadae2cad8f00